src/core/agent-manager/AgentManagerProvider.ts

```diff
@@ -0,0 +1,623 @@
+import * as vscode from "vscode"
+import { spawn, ChildProcess, execSync } from "node:child_process"
+import * as path from "node:path"
+import { fileExistsAtPath } from "../../utils/fs"
+import { AgentRegistry } from "./AgentRegistry"
+import { CliOutputParser, type StreamEvent, type KilocodeStreamEvent, type KilocodePayload } from "./CliOutputParser"
+import { getUri } from "../webview/getUri"
+import { getNonce } from "../webview/getNonce"
+import type { ClineMessage } from "@roo-code/types"
+
+const SESSION_TIMEOUT_MS = 120_000 // 2 minutes
+const WINDOWS = "win32"
+
+export function getKilocodeCliCandidatePaths(env: NodeJS.ProcessEnv, platform: NodeJS.Platform): string[] {
```

Comment:
I'm not sure about this implementation. Can't we use the `PATH` to look up `kilocode`? Since I use `mise-en-place` to install NodeJS, my `kilocode` bin is located at `/Users/remon/.local/share/mise/installs/node/22.15.1/bin/kilo`, and from the looks of it I'm not sure if this function will detect that.